src/main/java/uk/gov/justice/digital/delius/controller/secure/CourtAppearancesResource.java

```diff
@@ -47,5 +54,25 @@ public CourtAppearanceBasicWrapper getCourtAppearances(@ApiParam(name = "fromDat
         final var courtAppearanceBasics = courtAppearanceService.courtAppearances(fromDate.orElse(LocalDate.now()));
         return new CourtAppearanceBasicWrapper(courtAppearanceBasics);
     }
+
+    @ApiOperation(value = "Returns all court appearances associated with the CRN for the conviction ID.")
+    @ApiResponses(
+        value = {
+            @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
+            @ApiResponse(code = 401, message = "Unauthorised", response = ErrorResponse.class),
+            @ApiResponse(code = 403, message = "Forbidden", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "The offender CRN is not found", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error whilst processing request.", response = ErrorResponse.class)
+        })
+    @GetMapping(value = "/offenders/crn/{crn}/convictions/{convictionId}/courtAppearances")
+    public CourtAppearanceBasicWrapper getOffenderCourtAppearancesByCrn(final @PathVariable("crn") String crn,
+                                                                        final @PathVariable("convictionId") Long convictionId) {
+
+        List<CourtAppearanceBasic> appearances = offenderService.offenderIdOfCrn(crn)
```

Comment:
very minor; could use `final var`

Reply:
Done. Not that convinced about var when the right hand side doesn't explicitly show the typing. It's probably just the shock of the new :-)